rdbms/src/test/java/io/spine/server/storage/jdbc/query/SelectMessageByIdQueryShould.java

```diff
@@ -20,29 +20,117 @@
 public class SelectMessageByIdQueryShould {
 
     @Test
-    public void handle_database_exception() throws SQLException {
-        final SelectMessageByIdQuery query = Given.getSelectByIdQueryMock();
-        try {
-            query.execute();
-            fail();
-        } catch (DatabaseException expected) {
-            verify(Given.getLoggerMock()).error(anyString(), any(SQLException.class));
+    public void close_result_set() throws SQLException {
+        final SQLQuery underlyingQuery = mock(SQLQuery.class);
+        final ResultSet resultSet = mock(ResultSet.class);
+
+        doReturn(resultSet).when(underlyingQuery)
+                           .getResults();
+        final DataSourceWrapper dataSource = whichIsStoredInMemory(newUuid());
+        final ASelectMessageByIdQuery query = newBuilder().setTableName(newUuid())
+                                                          .setQuery(underlyingQuery)
+                                                          .setDataSource(dataSource)
+                                                          .build();
+        query.execute();
+        verify(resultSet).close();
+    }
+
+    @Test(expected = DatabaseException.class)
+    public void handle_sql_exception() throws SQLException {
+        final SQLQuery underlyingQuery = mock(SQLQuery.class);
+        final ResultSet resultSet = mock(ResultSet.class);
+
+        doReturn(resultSet).when(underlyingQuery)
+                           .getResults();
+        doThrow(SQLException.class).when(resultSet)
+                                   .next();
+        final DataSourceWrapper dataSource = whichIsStoredInMemory(newUuid());
+        final ASelectMessageByIdQuery query = newBuilder().setTableName(newUuid())
+                                                          .setQuery(underlyingQuery)
+                                                          .setDataSource(dataSource)
+                                                          .build();
+        query.execute();
+    }
+
+    @Test
+    public void return_null_on_deserialization_if_column_is_null() throws SQLException {
+        final ResultSet resultSet = mock(ResultSet.class);
+        final DataSourceWrapper dataSource = whichIsStoredInMemory(newUuid());
+        final Descriptors.Descriptor messageDescriptor = StringValue.getDescriptor();
+        final ASelectMessageByIdQuery query = newBuilder().setTableName(newUuid())
+                                                          .setMessageColumnName(newUuid())
+                                                          .setMessageDescriptor(messageDescriptor)
+                                                          .setDataSource(dataSource)
+                                                          .build();
+        final Message deserialized = query.readMessage(resultSet);
+        assertNull(deserialized);
+    }
+
+    private static class ASelectMessageByIdQuery extends SelectMessageByIdQuery<String, Message> {
```

Comment:
Let's have test queries like this one in `Given`s. And `Given` classes should belong to a dedicated packages, as you recall. 

In this way there will be no option to use any API except for `public`. Also, the `public` classes will have to be documented. No doubt, such an approach will increase the readability of tests.

Reply:
Done.